商品情報編集機能

tech-o-zukky opened the pull request:
レビューよろしくお願いします。
※商品購入機能関連は実装してないため、対応していません。
---
# 確認用の動画

[ログイン状態の出品者だけが商品情報編集ページに遷移できること](https://gyazo.com/104e621fee4e66ffea9277a67e33c9de)

※上記動画の続き
[必要な情報を適切に入力すると、商品情報（商品画像・商品名・商品の状態など）を変更できること](https://gyazo.com/99c8b83032559c5e4e0f365c4dd136dc)

---

[ログイン状態の出品者以外のユーザーは、URLを直接入力して出品していない商品の商品情報編集ページへ遷移しようとすると、トップページに遷移すること](https://gyazo.com/0a4248b1800ca766cb513b24273aae59)

---

[ログアウト状態のユーザーは、URLを直接入力して商品情報編集ページへ遷移しようとすると、ログインページに遷移すること](https://gyazo.com/eb2f54ced68e2c9fc7ee1246bf1b5547)

---
[動作確認](https://gyazo.com/5cc4d0b15e741da35638e6ad6834977d)

## 観点

- 商品名やカテゴリーの情報など、すでに登録されている商品情報は商品情報編集画面を開いた時点で表示されること（画像に関しては、表示されない状態で良い）
- 商品出品時とほぼ同じ見た目で商品情報編集機能が実装されていること
- 何も編集せずに更新をしても画像無しの商品にならないこと

---

[エラー時の挙動](https://gyazo.com/4014b7d3eb26d7909594b9ba215e1f71)
## 観点
- エラーハンドリングができていること（適切では無い値が入力された場合、情報は保存されず、エラーメッセージを出力させること）
- エラーメッセージの出力は、商品情報編集ページにて行うこと

tech-o-zukky committed:
ルーティング追加(edit, update)

tech-o-zukky committed:
コントローラーにアクション追加(edit, update)

tech-o-zukky committed:
初期表示用のデータ取得(edit)

tech-o-zukky committed:
DLしたビューファイル配置(edit)

tech-o-zukky committed:
ビューedit：hogeの修正

tech-o-zukky committed:
form_with修正

HTTPメソッドの誤り修正

tech-o-zukky committed:
コントローラーにupdate処理追加

tech-o-zukky committed:
もどるのパスを追加

「もどる」を押すと詳細ページに遷移

tech-o-zukky committed:
詳細から編集機能への遷移追加

tech-o-zukky committed:
Rubocop実行後の修正

tech-o-zukky committed:
editに合わせてコードを微修正

Katsuno0814 commented on app/controllers/items_controller.rb:
```diff
@@ -22,6 +23,19 @@ def create
+  end
+
+  def update
+    @item = Item.find(params[:id])
```
>@item = Item.find(params[:id])

こちら、複数のアクションに@item = Item.find(params[:id])を記載するのではなく、before_actionを使用してみてはいかがでしょう？理由は、同じコードを同ファイルで繰り返し書かないようにするためです。

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

tech-o-zukky committed:
レビュー後修正

・itemデータをbefore_actionで取得するよう修正（show, edit, update）
・上記に伴い、editだけで使用するログインチェック、出品ユーザーのチェックは、editメソッド内に移動。

tech-o-zukky commented:
確認ありがとうございます。
指摘を受けた点について修正し、修正後の動作確認をしました。
再レビューよろしくお願いします。

[商品詳細画面の表示](https://gyazo.com/396e5b74ec57e2ceb23032f64ac77eb8)

[詳細画面->編集画面->更新後の詳細画面](https://gyazo.com/cd2d24fce8233c70a714fbce884dd05b)

[ログイン状態の出品者だけが商品情報編集ページに遷移できること](https://gyazo.com/f5a3d2820f1db2c31a0386f13ae7ace6)

matsu987 reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！
